gradle/android-mixed-java-kotlin-project/AndroidSample/src/main/kotlin/org/jetbrains/kotlin/gradle/androidsample/MainActivity2.kt

```diff
@@ -15,17 +15,20 @@ open class MainActivity2 : Activity() {
         setContentView(R.layout.activity_main2)
 
         var next: Button = findViewById(R.id.Button02) as Button
-        next.setOnClickListener(object : View.OnClickListener {
-            public override fun onClick(view: View) {
-                val intent: Intent = Intent()
-                setResult(Activity.RESULT_OK, intent)
-                finish()
-            }
-        })
+
+//        next.setOnClickListener(object : View.OnClickListener {
```

Comment:
Yes, you're right. It works perfectly without the commented code. I've changed the app name (it wasn't displayed correctly in the app launcher - Activity name was taken instead). I've also added kotlin icon instead of the default Android one.